tests/components/gios/test_init.py

```diff
@@ -1,126 +1,120 @@
-    config_entry = MockConfigEntry(
-        domain=DOMAIN,
-        title="Home",
-        unique_id=123,
-        data={
-            "station_id": 123,
-            "name": "Home",
-        },
+    mock_config_entry.add_to_hass(hass)
+
+    device_entry = device_registry.async_get_or_create(
+        config_entry_id=mock_config_entry.entry_id, identifiers={(DOMAIN, 123)}
     )
 
-    indexes = json.loads(await async_load_fixture(hass, "indexes.json", DOMAIN))
-    station = json.loads(await async_load_fixture(hass, "station.json", DOMAIN))
-    sensors = json.loads(await async_load_fixture(hass, "sensors.json", DOMAIN))
-
-    with (
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_stations",
-            return_value=STATIONS,
-        ),
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_station",
-            return_value=station,
-        ),
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_all_sensors",
-            return_value=sensors,
-        ),
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_indexes",
-            return_value=indexes,
-        ),
-    ):
-        config_entry.add_to_hass(hass)
-
-        device_entry = device_registry.async_get_or_create(
-            config_entry_id=config_entry.entry_id, identifiers={(DOMAIN, 123)}
-        )
-
-        await hass.config_entries.async_setup(config_entry.entry_id)
-        await hass.async_block_till_done()
-
-        migrated_device_entry = device_registry.async_get_or_create(
-            config_entry_id=config_entry.entry_id, identifiers={(DOMAIN, "123")}
-        )
-        assert device_entry.id == migrated_device_entry.id
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    migrated_device_entry = device_registry.async_get_or_create(
+        config_entry_id=mock_config_entry.entry_id, identifiers={(DOMAIN, "123")}
+    )
+    assert device_entry.id == migrated_device_entry.id
 
 
 async def test_remove_air_quality_entities(
-    hass: HomeAssistant, entity_registry: er.EntityRegistry
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    mock_config_entry: MockConfigEntry,
+    mock_gios: MagicMock,
 ) -> None:
     """Test remove air_quality entities from registry."""
+    mock_config_entry.add_to_hass(hass)
     entity_registry.async_get_or_create(
         AIR_QUALITY_PLATFORM,
         DOMAIN,
         "123",
         suggested_object_id="home",
         disabled_by=None,
+        config_subentry_id=mock_config_entry.entry_id,
     )
 
-    await init_integration(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
 
     entry = entity_registry.async_get("air_quality.home")
     assert entry is None
+
+
+async def test_unique_id_migration(
```

Comment:
I moved this code to `test_init.py` because its main logic is in the `__init__.py` file. Additionally, we have a global `init_integration` fixture set in the `test_sensor.py` file.